web/modules/custom/bnf/src/Plugin/FieldTypeTraits/DateTimeTrait.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Drupal\bnf\Plugin\FieldTypeTraits;
+
+use Drupal\bnf\GraphQL\Operations\GetNode\Node\Paragraphs\HeroDate\DateTime;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Spawnia\Sailor\ObjectLike;
+
+/**
+ * Helper trait, for dealing with DateTime fields.
+ */
+trait DateTimeTrait {
+
+  /**
+   * Getting Drupal-ready value from object.
+   *
+   * @return mixed[]
+   *   The value that can be used with Drupal ->set().
+   */
+  public function getDateTimeValue(DateTime|ObjectLike|null $dateTime, bool $includeTime = TRUE): array {
+    if (is_null($dateTime)) {
+      return [];
+    }
+
+    /** @var \Drupal\bnf\GraphQL\Operations\GetNode\Node\Paragraphs\HeroDate\DateTime $dateTime */
+    $date = DrupalDateTime::createFromTimestamp($dateTime->timestamp);
+    $date->setTimezone(new \DateTimeZone($dateTime->timezone));
+
+    return [
```

Comment:
WTF? The `datetime` fields `DateTimeItem` doesn't have any handy setters so one can simply do `$node->field_somedate->set(\DateTime)` (or at least the `DrupalDateTime` class)? For crying out loud.

Reply:
You mean the mess about `$includeTime`? Because yeah - that is a bit of a clusterfuck, that a Drupal DateTime field can return different things..

Reply:
Well, sorta... Meant the fact that you can't do `$node->field_somedate->set(new \DateTime());`, or something equivalent that works with proper date objects, but have to *provide a formatted string* (without a timezone even).

Reply:
Yeah, it's a mess.